### DISCUSSION TITLE
Formatter Definition

### DISCUSSION THREAD

**Discussion by fbalzarotti** [ImSwitch/ImSwitch#203] [General]
Hi, 

It looks to like there has not been a definition of a code formatter for ImSwitch.

I'm bringing this up because each ImSwitch file that I touch gets heavily formatted.

In my group we use VSCode as IDE and YAPF as formatter automatically on save. 
I'm aware it's not the most common choice and we might even change to e.g. black.

https://github.com/google/yapf 
https://github.com/psf/black

Has there been a conversation about this? 

@beniroquai, @jacopoabramo, @GuillaumeMinet 
or are you already fixed on a formatter?

**Comment by jacopoabramo** [comment]
Hi @fbalzarotti, I think we did not have a serious discussion about using a proper code formatter but I would be happy to see it. I'm not a big fan of black but I think it's one of the most widely used as you suggested, so I would be ok with it. How would you like to proceed?

**Reply by fbalzarotti** [reply]
I see, thanks @jacopoabramo. In that case, I propose a bunch of changes here https://github.com/ImSwitch/ImSwitch/discussions/204
